futurenhs.app/pages/groups/[groupId]/members/[memberId]/update/index.page.tsx

```diff
@@ -0,0 +1,171 @@
+                            context,
+                            routeParams.GROUPID
+                        )
+                        const memberId: string = selectParam(
+                            context,
+                            routeParams.MEMBERID
+                        )
+
+                        const form: any =
+                            props.forms[formTypes.UPDATE_GROUP_MEMBER]
+
+                        /**
+                         * Get data from services
+                         */
+                        try {
+                            const [memberData, groupRoles] = await Promise.all([
+                                getGroupMember({ groupId, user, memberId, isForEdit: true }),
+                                getGroupRoles({ groupId, user })
+                            ])
+                            const etag = memberData.headers?.get('etag');
+                            props.etag = etag
+
+                            props.member = memberData.data
+                            props.layoutId = layoutIds.GROUP
+                            props.tabId = groupTabIds.MEMBERS
+                            props.pageTitle = `${props.entityText.title} - ${props.member.firstName ?? ''
+                                } ${props.member.lastName ?? ''} - Edit`
+
+                            /**
+                             * Handle setting role options for multi-choice 
+                             */
+                            const roleOptions: Array<FormOptions> = groupRoles?.data?.map((role) => {
+                                return {
+                                    value: role.id,
+                                    label: role.name
+                                }
+                            })
+
+                            const updatedRolesForm = setFormConfigOptions(form, 0, 'groupUserRoleId', roleOptions);
+                            const usersCurrentRole = groupRoles.data?.find((role) => role.name === props.member?.role)
+
+                            props.forms[formTypes.UPDATE_GROUP_MEMBER] = updatedRolesForm;
+
+                            updatedRolesForm.initialValues = {
+                                groupUserRoleId: usersCurrentRole?.id,
+                            }
+
+
+
+                            /**
+                             * Handle server-side  form post
+                             */
+                            if (
+                                currentValues &&
+                                requestMethod === requestMethods.POST
+                            ) {
+
+                                const headers =
+                                getStandardServiceHeaders({
+                                    csrfToken,
+                                    etag,
+                                });
+
+                                if (currentValues.body?.['_form-id'] === updatedRolesForm.id) {
```

Comment:
This feels like a thing to abstract/ avoid typos using a more generic selector

e.g. isFormType(currentValues, updatedRolesForm.id)

Reply:
Made a helper for this and now checking if (isRoleForm) / (isDeleteForm)